### PR TITLE
Add an HTML5 color input

### PR DESCRIPTION
**Pull request by delehef**
Hello,

I needed a color input for a project of mine and didn't find any in Phoenix, so I added one manually. Here is the PR if you wish to merge it.

Thanks for your work!

### CODE REVIEW

**Comment by chrismccord** [comment]
Thanks for the PR!. I didn't know this was a thing. How supported is this in browsers today? @josevalim wdyt?

**Comment by josevalim** [comment]
We also need tests, thank you.

**Comment by tmjoen** [comment]
Not supported in IE and Safari, http://caniuse.com/#feat=input-color

**Comment by delehef** [comment]
> We also need tests, thank you.

My bad, I just added the same than the other input fields.

File: test/phoenix_html/form_test.exs [hunk 1]
```diff
@@ -252,8 +252,26 @@ defmodule Phoenix.HTML.FormTest do
...
+    assert safe_to_string(color_input(:search, :key, value: "#123456", id: "key", name: "search[key][]")) ==
+          ~s(<input id="key" name="search[key][]" type="color" value="#123456">)
+  end
+
+  test "color_input/3 with form" do
+    assert safe_form(&color_input(&1, :key)) ==
+          ~s(<input id="search_key" name="search[key]" type="color" value="value">)
+
+    assert safe_form(&color_input(&1, :key, value: "foo", id: "key", name: "search[key][]")) ==
+          ~s(<input id="key" name="search[key][]" type="color" value="foo">)
```
Comment (josevalim) [inline test/phoenix_html/form_test.exs]: This input is wrong I think.

File: test/phoenix_html/form_test.exs [hunk 2]
```diff
@@ -252,8 +252,26 @@ defmodule Phoenix.HTML.FormTest do
     assert safe_form(&search_input(&1, :key)) ==
           ~s(<input id="search_key" name="search[key]" type="search" value="value">)
 
-    assert safe_form(&search_input(&1, :key, value: "foo", id: "key", name: "search[key][]")) ==
-          ~s(<input id="key" name="search[key][]" type="search" value="foo">)
+    assert safe_form(&search_input(&1, :key, value: "#123456", id: "key", name: "search[key][]")) ==
+          ~s(<input id="key" name="search[key][]" type="search" value="#123456">)
```
Comment (josevalim) [inline test/phoenix_html/form_test.exs]: You changed this input without a need. :)

**Comment by delehef** [comment]
You're right, it should be OK now.

**Comment by josevalim** [comment]
Can you please squash it all now? Then we are ready to go! :D

**Comment by delehef** [comment]
First time I squash commits together :s 

Is it correct?

**Comment by josevalim** [comment]
No, it is not. :) We have 7 commits now and it should be one. Would you like me to take care of it this time?

**Comment by delehef** [comment]
Yes please, and would you mind showing me how you did after?

**Comment by josevalim** [comment]
I will pretty much do this:

http://stackoverflow.com/questions/2563632/how-can-i-merge-two-commits-into-one

You just need to use HEAD~N where N will lead you to the first commit that is not yours. Than you pick the first one you have and squash all the rest. Once you are done:

    git push -f origin master

And the code here will be up to date.

Do you want to give it another try?

**Comment by delehef** [comment]
Ah, I think it's OK now?

I did a normal `push` before instead of a `push -f`.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by josevalim** [comment]
Perfect, thank you!

**Comment by delehef** [comment]
Nice :)

**Pull request merged**